tools/profileBuilder/latest.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+// Copyright 2017 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/marstr/collection"
+)
+
+// LatestStrategy evaluates the Azure-SDK-for-Go repository for the latest available API versions of each service.
+type LatestStrategy struct {
+	Root      string
+	Predicate func(packageName string) bool
+}
+
+// AcceptAll is a predefined value for `LatestStrategy.Predicate` which always returns true.
+func AcceptAll(name string) (result bool) {
+	result = true
+	return
+}
+
+// IgnorePreview searches a packages "API Version" to see if it contains the word "preview". It only returns true when a package is not a preview.
+func IgnorePreview(name string) (result bool) {
+	matches := packageName.FindStringSubmatch(name)
+	if len(matches) >= 4 {
```

Comment:
>>= 4 [](start = 17, length = 4)

These number look a bit magical, add a comment explaining a little bit.

Reply:
You're telling me you don't appreciate the magic in life? Sure thing, I'll add some documentation here.

I was actually quite frustrated with Go's Regexp library, because in C# you can reference groups by the name you gave them instead of some index. The way goes about it, it's quite prone to these sorts of magic numbers.

Reply:
Fixed locally